File: ws/src/main/scala/de/thm/ii/fbs/services/checker/trait/CheckerServiceOnChange.scala
package de.thm.ii.fbs.services.checker.`trait`

import de.thm.ii.fbs.model.{CheckrunnerConfiguration, Task}

trait CheckerServiceOnChange {
  def onCheckerConfigurationChange(task: Task, checkerConfiguration: CheckrunnerConfiguration,
                                   mainFile: String, secondaryFile: String): Unit
}

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/services/v2/checker/trait/CheckerServiceOnChange.kt
package de.thm.ii.fbs.services.v2.checker.trait

import de.thm.ii.fbs.model.v2.CheckrunnerConfiguration
import de.thm.ii.fbs.model.v2.Task

interface CheckerServiceOnChange {
    fun onCheckerConfigurationChange(task: Task, checkerConfiguration: CheckrunnerConfiguration)
}